### ISSUE TITLE
GCM bitcode errors

### ISSUE THREAD

**Issue by code-smith** [phonegap/phonegap-plugin-push#446]
I'm getting lot of errors with google gcm and i know the solutions is to disable bitcode in  X-Code.
But i'm getting too many warnings and might end up submitting an app which apple might delay in approving.
Is there any way to disable all this GCM clutter for IOS?

**Comment by macdonst** [comment]
@code-smith not currently. What I want to do is to move the GCM stuff on iOS to a cocoapod but I need to get Cordova/PhoneGap to support it as a framework first.

**Comment by code-smith** [comment]
Ok, but I really think including gcm is very messy as its giving all these build warnings..and we need to disable bitcode. 
@macdonst Also have you every come across any problem with distribution when bitcode is disabled?I'm pretty worried about this behavior. Please do help out

**Comment by macdonst** [comment]
@code-smith No, I haven't run into a problem with Apple with a bitcode disabled app. In fact we just got approval for one last week.

I fully admit that the GCM code in iOS is messy, I even mention that in the issue itself. I have a plan to clean it up but it will require some changes to Cordova/PhoneGap as well. However GCM on iOS was requested a bunch of times so I wanted to enable people to start using it.

**Comment by code-smith** [comment]
@macdonst  You just released a lot of stress that was building up. Thanks for answering.
And some really brilliant work you guys are doing. keep it up man. good luck ahead

I wrote myself some code for using parse push notification and ran in to some problem and raised another issue i raised, #447.

I dont understand the point of the method notification_applicationDidBecomeActive, when its not getting called.

**Comment by code-smith** [comment]
@macdonst  i dont know how to leave a personal message to you , please consider this as pm .
This is the first time i'm uploading an app to istore, should i disable bitcode auto-recompilation or leave it as such ?

**Comment by macdonst** [comment]
@code-smith if your app isn't compiling with bit code enabled then turn it off. This SO post is much better at explaining it than I would ever be. Check the first 2 answers:

http://stackoverflow.com/questions/31088618/impact-of-xcode-build-options-enable-bitcode-yes-no

**Comment by code-smith** [comment]
I did disable bitcode in xcode. 
There is an option in itunes connect to disable "bitcode auto-recompilation".
As i have already disabled it in xcode and continued building form there on.. i think it makes sense for me to disable in itunes connect also, but i wasn't sure , i was referring to this options in I-tunes reconnect, not the one in x-code..
sorry for the confusion

**Comment by flashnet69** [comment]
I have the same problem when i try to build on my iPhone device (Simulator OK)...

...Plugins/phonegap-plugin-push/GCM/Libraries/libGGLCloudMessaging.a(GGLContext+CloudMessaging.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. for architecture armv7

@macdonst As I understand it , you can not correct this problem because the solution is not simple ?

**Comment by macdonst** [comment]
@flashnet69 Just turn bitcode off. I need a new lib from Google in order to fix that bitcode issue.

**Comment by gouroujo** [comment]
I don't if it's could be a solution but maybe you can delete all the GCM related files and reference in the generated Xcode project when you build iOS?

**Comment by panaggio** [comment]
If [this](https://github.com/google/gcm/issues/91) gets fixed, this issue could be more easily fixed, right?

**Comment by panaggio** [comment]
Besides that, could I help in some way here @macdonst? I've written plugins in the past (actually, I think the first and last plugin I wrote were push plugins :)), and I've patched my own cordova a few times (back on v0 and v1). If there's anything I could do to help here, I'm all ears(?/eyes?).

I have an upcoming release window (next week) where I need to release action button support on Apple Watch (so I need bitcode support) and I have two options:

* help this get fixed at about next week's Tuesday or
* remove GCM support and ship a forked version of the plugin

Unless you tell me there isn't a chance that we can get it done in ~7 days, I still think it's worth a try :) It seems easy to just remove GCM, but it's a very worthless use of time, because we could use that same (probably more) effort to really get this fixed. Worst case, I can rush to remove GCM on Monday if we can't reach estimated goals until then.

**Comment by macdonst** [comment]
@panaggio yeah, once google/gcm#91 is fixed then this problem will go away. 

Sorry, I have been out of touch as I was on vacation. If you have already gone and removed GCM in a fork please let me know via a PR to a new branch. I'm thinking of maintaining a branch with out GCM iOS support as it causes problems for a bunch of people.

**Comment by panaggio** [comment]
> Sorry, I have been out of touch as I was on vacation.

Np.

> If you have already gone and removed GCM in a fork please let me know via a PR to a new branch. I'm thinking of maintaining a branch with out GCM iOS support as it causes problems for a bunch of people.

I', not there yet. I'm having some issues, and AFAICS it isn't on the fork itself, but it could be. As soon as the fork works properly, I'll create the PR.

Anyway, I'm still up to helping fix the plugin. Let me know if I can be of help.

**Comment by bjakubiak** [comment]
https://github.com/google/gcm/issues/91 has been fixed 2016-04-22. Is it now enough to increase phonegap-plugin-push version?

**Comment by macdonst** [comment]
@bjakubiak it got fixed but I don't see an update on CocoaPods. The latest version is from back before the bug was fixed.

**Comment by arthuralves** [comment]
@macdonst have any release date for this fix?

**Comment by macdonst** [comment]
@arthuralves last cocoapod release of GCM was in February. Waiting for Google at this point:

https://cocoapods.org/?q=GoogleCloud

**Comment by NickAdriaenssens** [comment]
@macdonst any update on this? Thanks.
